### DISCUSSION TITLE
[非官方] Issue & Discussion 投递须知

### DISCUSSION THREAD

**Discussion by SALTWOOD** [Meloong-Git/PCL#2663] [讨论]
这里是盐木。如题，这是一个**非官方**的 Issues 投递注意事项。
如果你想要提交一个 Issue ，这篇文章可以作为参考。
注：**此文章部分为个人观点，有错误请指出。**

### 🔍快速寻找类似的 Issue

首先，判断你的问题是否重复，并且在 **[【持续更新】常见&难检反馈及问题列表](https://github.com/Hex-Dragon/PCL2/discussions/1930)** 处搜索并确认其中没有你所遇到的问题后，即可进入下一步——在 [Issues 页面](https://github.com/Hex-Dragon/PCL2/Issues)搜索你的问题。

在 [Issues 页面](https://github.com/Hex-Dragon/PCL2/Issues)检索类似的 Issue 时，不要直接将你遇到的问题丢进搜索框来搜索，因为个人的表达问题，同一个问题由多个人来描述可能会有截然不同的效果。你应该拆分“关键词”（如例）：

遇到的问题：整合包安装时选择是否安装可选的模组
搜索：“整合包安装时选择是否安装可选的模组” ❌
搜索：“可选模组 整合包 安装” ✔

如果这个问题看起来不太像是启动器的问题，你也可以在 [Discussions 页面](https://github.com/Hex-Dragon/PCL2/discussions)先提出你的问题，然后在有人给出建议之后再考虑是否提交 Issue。

### 🧭选择 Issue 的类型

目前 PCL2 仓库的 Issue 大致分为三种类型——“**Bug**”、“**优化**”和“**新功能**”三种：

“**Bug**”：这类型的 Issue 通常是在使用 PCL2 或是启动 Minecraft 的过程中遇到问题。
“**优化**”：这类型的 Issue 通常是对 PCL2 的现有功能的优化（即小幅度调整）
“**新功能**”：这类型的 Issue 通常是希望在 PCL2 中添加一个新的功能，或是对某个功能的大幅度修改（相当于重做的那种）

**Bug 反馈**通常需要提交 **PCL2 日志**或是 **Minecraft 错误报告**（是一个压缩包，**而不是此页面的截图！**），除非这个问题不起眼到 PCL2 的日志都没记载它。

### 🤔取一个好的标题

一个好的标题可以帮助开发者快速了解 Issue 的大意，并以此来判断此问题的处理优先级、处理难度，在一定程度上还可以帮助其他反馈者更方便地检索类似的问题并以此判断自己要反馈的问题是否已经重复。下面是一些可参考的标题：

**“[BUG] Runtime Invoke 线程执行失败”** ✔，此标题很好地描述了**遇到的问题**，并且标示了 Issue 类型。
**“辅助功能：默认设置为中文，请勿修改forceUnicodeFont”** ✔，此标题描述了现状并且提供了优化建议。
**“希望龙猫给pcl添加签名证书”** ✔，这个标题明确地表示出了对某优化的具体信息。
**“发电”** ❌，这个标题**什么信息都没有**，~甚至原 Issue 就是一个发癫的 Issue~。
**“Minecraft 崩溃”** ❌，这个标题概括性过于强，以至于**什么都判断不出来**。
**“memory clear”** ❌，标题什么信息都没体现，同时二义性过强，可能会被理解错误。

### 📃撰写文章的正文部分

正文部分除非必要，否则不要写得太长。过长的正文会增加开发者的阅读成本，还会因为各种原因导致其无法作出 Issue 的定性。正文部分尽量言简意赅，将报错、问题描述写清楚即可，有能力还可以提出可能的解决方案，或是大概提出一个修复思路。

### ⚠在提交 Issue 时的其他注意事项

| 注意点 | 注意事项 |
|----------|:-------------:|
| 上传日志 | 上传日志时，先点击日志上传区，然后将 PCL2 的日志或是 Minecraft 错误报告压缩包拖拽至那里。</br>**尤其注意！尤其是像压缩包之类上传需要一定时间的文件，等待文件彻底上传完毕——即“[Uploading 某某某...]”字样消失时才提交 Issue，否则会上传不完整导致开发者需要等待你补充日志！** |
| 检查项 | 请认真地逐个检查检查项列表中的信息，并且记得在检查完毕后勾选左侧的勾选框。</br>**不要看都不看就全部勾上！** |

暂时就想到那么多，如果有需要补充的可以@我

此文章的贡献者：@SALTWOOD、@Chiloven945

**Comment by so1ve** [comment]
🤔你觉得有人会在开issue之前先来看这篇文章吗（）我建议给issue template提交一个PR，但是意义估计不大

**Reply by iPodtouch6557** [reply]
其实issue template里加这玩意意义不大，因为以后这样的discussions多一点那个模板得臃肿死……
所以在此也建议不要再提交相关内容的了，如有建议可以去已有的相关话题下面向原作者建议更改

**Reply by MoYuan-CN** [reply]
丢到 Issue 选择页面去？

**Comment by Chiloven945** [comment]
希望、请求这类的祈使句词实际上可以不用，因为你有需求才会来发送issue所以说可以省略，否则看起来有点冗余，其次标题也应该使用陈述句进行表达
人名也大可不必，因为你交issue就是给开发者看的……
还有就是是描述与issue无关的，一些完全没有存在必要的修饰词，比如说不小心、意外

**Reply by SALTWOOD** [reply]
正在修改

**Comment by HomoobyDX** [comment]
因为我当时乱交issue，以至于屏蔽一个月，可能发了这个是为了提醒部分（像我这样的）人不要遇到什么问题就发

**Comment by WorldHim** [comment]
~~太长不看~~
- 篇幅太长，建议只保留必要部分（删去前两段）；
- 拼写不规范，Issue 统一大小写；
- 基本事实错误，评判一个问题是否提交不是以常见度，而是以与启动器关系程度（是否是启动器本身的问题而非游戏问题，若不使用本启动器下同样问题则不应该提交）。

**Reply by SALTWOOD** [reply]
此处“常见”的意思是比较多人关心的问题，比如 Quilt 安装
用词不当，已改为“重复”
其他也已经修改

**Comment by MoYuan-CN** [comment]
其实 在 #1930 中有一段折叠起来的 Issue 提交规范，那段写的就很好
![image](https://github.com/Hex-Dragon/PCL2/assets/109946426/a746d556-fe12-4d1c-849f-5e2e5d628e78)
参考那段，并加上 Discussion 部分把

**Reply by SALTWOOD** [reply]
ok，会修改

**Comment by MoYuan-CN** [comment]
链接部分的后面最好加个空格 
~~貌似只有【持续更新】... 一个链接后面加了空格~~
